core/src/main/java/org/apache/accumulo/core/data/Mutation.java

```diff
@@ -962,6 +965,23 @@ public QualifierOptions family(Text colFam) {
       return family(colFam.getBytes(), colFam.getLength());
     }
 
+    /**
+     * Sets the column family, column qualifier, and column visibility of a mutation.
+     *
+     * @param key key
+     * @return a TimestampOptions object, advancing the method chain
+     */
+    @Override
+    public TimestampOptions keyColumns(Key key) {
+      Objects.requireNonNull(key, "key cannot be null");
+
+      Text colFam = key.getColumnFamily();
+      Text colQual = key.getColumnQualifier();
+      Text colVis = key.getColumnVisibility();
```

Comment:
Creating a hadoop Text object will copy the byte array in the key object.  The following code will not copy the byte array in the key object.

```suggestion
      byte[] colFam = key.getColumnFamilyData().toArray();
      byte[] colQual = key.getColumnQualifierData().toArray();
      byte[] colVis = key.getColumnVisibilityData().toArray();
```